### DISCUSSION TITLE
How I can uninstall a KongPlugin/KongClusterPlugin

### DISCUSSION THREAD

**Discussion by shahamit** [Kong/kubernetes-ingress-controller#1834] [Q&A] [answered]
I have installed multiple kong plugins - oauth2, rate-limiting through kubernetes yaml files. I can see them through `kubectl get KongPlugins -n kong`

How can I uninstall them?

Thanks.

**Comment by shaneutt** [comment, marked as answer]
Hi @shahamit,

If these were originally deployed with manifest files (e.g. `kubectl apply -f plugins.yaml`) you can delete them each by name:

```console
$ kubectl -n kong delete kongplugin <name>
```

However keep in mind that operation is permanent.

If you deployed these plugins some other way (such as using some management system like `helm` with your own personal helm chart) then deleting them directly may not be correct and you would want to figure out the correct cleanup methods for your deployment tool.

**Reply by shahamit** [reply]
Thanks. I had deployed the plugins through the manifest files hence the above command worked seemlessly.

**Comment by ccfishk** [comment]
Right, second Shane's suggestion, aslo helm charts command
helm uninstall *** -n name-space could be helpful.
